### ISSUE TITLE
CRITICAL: Amazon Drive does not work anymore with rclone 429: "429 Too Many Requests" / Rate exceeded 

### ISSUE THREAD

**Issue by ajkis** [ncw/rclone#1417]
It seems Amazon Drive blocked rclone, I tested it on 4 different servers, tried reauth the app but no success.

Any rclone command will deliver the following errors:
```
2017/05/18 11:19:14 DEBUG : pacer: Rate limited, sleeping for 666.145821ms (1 consecutive low level retries)
2017/05/18 11:19:14 DEBUG : pacer: low level retry 1/10 (error HTTP code 429: "429 Too Many Requests": response body: "{\"message\":\"Rate exceeded\"}")
```

**Comment by Technifocal** [comment]
@ncw Already seems to be aware, https://github.com/ncw/rclone/issues/1415#issuecomment-302337151

**Comment by PhilipFraDIKU** [comment]
Yes, getting the exact same problems. When typing `rclone lsd ACD:ACD_Encrypted` in terminal I get:

`2017/05/18 11:24:30 Failed to create file system for "ACD:ACD_Encrypted": failed to get endpoints: HTTP code 429: "429 Too Many Requests": response body: "{\"message\":\"Rate exceeded\"}"`

**Comment by l3uddz** [comment]
If rclone is indeed banned like acd_cli, they can kiss my subscription goodbye (not that they care). This is ridiculous. Perhaps they dont like that rclone security profile keys can be accessed by anyone, as evidenced by those guys trying to impersonate rclone with acd_cli. We must wait to hear what @ncw says as anything else is hearsay.

**Comment by mdubash** [comment]
I had a transfer in operation at 01:56 this morning (UK BST) when the logs show the transfer just stopped - it went to 0 bytes/s and stayed there. rclone lsd produces the exact same message as the OP.

**Comment by ncw** [comment]
I have some emails out to contacts at Amazon asking what has happened.  They are on the west coast of the US I think, so they won't be in to work for a few hours...

@mdubash interesting info - thanks

**Comment by Joshua1337** [comment]
You can still get authcodes btw...

**Comment by geekcroft** [comment]
@ncw have you seen the forum post from the guy who had a reply from Amazon DE?

**Comment by ncw** [comment]
@geekcroft yes I have.  I'd like confirmation though.

**Comment by geekcroft** [comment]
@ncw Yeah am hoping its just some 2-bit support guy not knowing his stuff :(

**Comment by ajkis** [comment]
I also wanted to check for my self and called customer support just now, they told me they got notification today that rclone wont be supported anymore and that I should use their Desktop App.

This is really beyond retarded, @ncw I hope you guys can deliver local cache soon so we can use gdrive instead. 

I just switched all my servers to plexdrive until rclone gets local cache.

p.s. Its really shame that multipurpose tools like rclone gets removed from Amazon eg all apps with big traffic will get removed eventually.

**Comment by Technifocal** [comment]
@ajkis If this is true, I will have to move over to GDrive. :(

**Comment by scriptzteam** [comment]
F*UCK OFF AMAZON, I KNOW YOU READ, FUCK OFF _!_

**Comment by thenoahcomputer** [comment]
Wow this is seriously bad timing for me.  I was in the process of restoring a backup when acd_cli got disabled, switched to rclone and now it's disabled too. I've emailed Amazon about it, but don't expect much from them given their recent blatant disregard for their paying customers need to access their own data.

Anyone know of another way to mount or pull files out of ACD with linux or shuffle it over in bulk to gdrive?

**Comment by Aldigator** [comment]
An employee of Amazon Germany just told me on the phone that they are making changes to their services and infrastructure and that rclone probably needs to be updated accordingly.

**Comment by ajkis** [comment]
> An employee of Amazon Germany just told me on the phone that they are making changes to their services and infrastructure and that rclone probably needs to be updated accordingly.

I highly doubt that they would instruct their Tehnical staff to tell customer that rclone access is revoke and it would not be enabled again. Thats what they told me and i was on phone for more then 20 minutes asking if its possible it would work in the future since my usage is strictly rclone depended. 

They put me on hold for 5 minutes ( after i was first told that rclone access is revoked ) then he told me that rclone access wont be renabled for sure and that i need to find another way to use Amazon Drive ( since i told them i need linux client that they dont support )

**Comment by dafal** [comment]
This offer from the French provider ikoula may help : https://express.ikoula.com/en/serveur-dedie
More or less 17 €/month for the first month (or the first 3 months) for a Synology with a 4Tb disk.  Synology Cloud Sync can do the job.  Limited to 100 Mb/s

**Comment by geekcroft** [comment]
@ncw anything on the Email front yet mate?

**Comment by nbyloff** [comment]
@thenoahcomputer I am currently using these services to move data from ACD to Google
https://www.cloudhq.net
https://www.multcloud.com

**Comment by thenoahcomputer** [comment]
I called Amazon.  They were very friendly and understanding, but they said it's going to take them 2 days to get back to me with a solution.

@nbyloff 
Thanks for the info.  Looks like a much faster solution than waiting on Amazon.

**Comment by ginsu0** [comment]
@nbyloff I started a transfer a few hours ago using both of those, but neither is exceeding 0.5MB/s. At that rate, my transfer will take 6 months. Looking for another solution, but haven't found anything good so far.

**Comment by mdubash** [comment]
@thenoahcomputer
You got a better response than I did. Which was: we don't support anything other than our own apps as everything is always changing at the server end. Fingers crossed.

**Comment by M0V3** [comment]
@ginsu0 
Have a look at this https://www.reddit.com/r/AlienBlue/comments/1i13io/subreddit_links_r_and_username_links_u_are/

**Comment by ginsu0** [comment]
@M0V3 Are you sure you linked the right page?

**Comment by M0V3** [comment]
@ginsu0 Yea, edited my post.
Don't know how I managed to copy the wrong link.

Here it is one more time https://www.reddit.com/r/DataHoarder/comments/6bwbst/how_would_you_transfer_8tb_of_linux_isos_from_acd/dhpzcnn/

**Comment by thenoahcomputer** [comment]
I probably got better response because I didn't ask for support for rclone. I told them I just want my data back and can't get at it, because they disabled the only apps that would let me do that on Ubuntu.  They suggested the web interface, which I told them would just spin it's wheel and then not download anything.

It'd be ideal if they'd offer a read-only S3 bucket without data charges to get my stuff back so I can just rclone it to Gdrive. The multcloud.com site seems to be working, but significantly slower than rclone.

**Comment by ginsu0** [comment]
@M0V3 Thanks. If that's what it's come down to, I'll spin up a Windows VM with the native client and move a few directories at a time. It's a shame there isn't a better way.

**Comment by kiding** [comment]
#1409 #1415 have basically converged into the same issue after all. Leaving a comment for reference's sake.

**Comment by uncuervo** [comment]
Hi guys,

I'm a newbie on this community so please be kind to me if I'm not compliance 100% with any terms or conditions that apply here :P 

The point is;

I've been using rclone for a while (great app btw) just to connect my linux to ACD and encrypt on the fly, but such a nasty surprise I've got after seeing my ACD access with rclone was not working anymore...! maybe my fault? I was hoping so.. but NO!! it gets worse after seing all of this happened due all of the monkey business amazon is carrying out...? stating to "be open" and at the same time pushing us to do things exactly how they want (no blame, is their service but at least be honest!).

Anyway, I don't now for how far... but looks like amazon drive is not accepting more developers?

https://developer.amazon.com/amazon-drive

I'm saying that because despite of how complicated it can be, after checking the current situation I was considering (also as a suggesiton to everyone) to create our own apps so they can't ban any specific one due many users using it, or things like that... but, no luck going that way after finding that they've locked new registrations :/

Which drives me to; @ncw !!!, I have faith on you :) (sure many do)... and I'm sure after seeing such great job all of you have done so far you will be able to figure out how to address this issue. I can read on some comments that amazon support is saying they are doing changes and developers will need to adapt their apps? is that true?

I hope to hear from you guys.

All the best,

**Comment by gryphonmyers** [comment]
The funny thing about their "use our sync client" response is that I tried, I really did, and it is utter garbage. It made my CPU Max out constantly and constantly forgot what it was doing, forcing me to reauth. POS client for an increasingly POS service

**Comment by barnabyc** [comment]
For reference I'm seeing a slightly different message right now stating 401 Unauthorized and "invalid_client":
```
2017/05/18 22:35:07 Failed to create file system for "remote:foobar": failed to get endpoints: Get https://drive.amazonaws.com/drive/v1/account/endpoint: oauth2: cannot fetch token: 401 Unauthorized
Response: {"error_description":"Client authentication failed","error":"invalid_client"}
```

**Comment by calisro** [comment]
Yes. If look in the manage apps section in cloud drive, rclone is no longer listed just like acdcli was removed. This looks like permanent or at least long term ban now....

**Comment by niranjanshr13** [comment]
@calisro is there is anyway to create own token?

**Comment by left1000** [comment]
anyone know where the appropriate place is to contact ACD support to beg for a positive outcome is?

**Comment by mc40638** [comment]
I recently started using rclone and am dismayed at this turn of events.  Please keep us informed.

**Comment by niranjanshr13** [comment]
hey guys FolderSync is working with amazon cloud drive, FolderSync is android app to upload, download or sync. It must be a token problem.

**Comment by ajkis** [comment]
> hey guys FolderSync is working with amazon cloud drive, FolderSync is android app to upload, download or sync. It must be a token problem.

Yes they revoke/banned rclone token

**Comment by boblatino** [comment]
Moving to backblaze. They do have a 5 USD unlimited subscription and that is almost the same as ACD but they actually want developers to use their platform F@#$ off amazon. I do not know if the b2 integration does work with the "personal backup" but if it does not I will send a PR with it. Long live to rclone

**Comment by ajkis** [comment]
> Moving to backblaze. They do have a 5 USD unlimited subscription and that is almost the same as ACD but they actually want developers to use their platform F@#$ off amazon.

What does this mean ? is it viable option to replace ACD at least for cold storage ( using rclone)

**Comment by SirJMD** [comment]
The BackBlaze is limited to only backup your computer. As far as I can see it only works with their sh*tty desktop app.

**Comment by left1000** [comment]
"
Please understand that, At this time, the Amazon Drive ended the invitation period for new third party apps. That's the reason why you're unable to connect to your cloud drive using  rclone.

However, I have forwarded your email as feedback to our Amazon Drive Feedback & Development Team to look into this. They will surely consider your feedback while making improvement to our services in future.
"


what? so this has nothing to do with excessive usage or security? or rclone or acd_cli? and everyone single 3rd party api is being shut down? or was this support staff member just clueless and thought I was trying to get my own api token and that's what wasn't working? that would really suck as it would mean they won't let anyone ever come up with any functional api app.

**Comment by ncw** [comment]
I put a [post in the forum](https://forum.rclone.org/t/rclone-has-been-banned-from-amazon-drive/2314) which I'll update as and when I hear more info.

**Comment by SirJMD** [comment]
> what? so this has nothing to do with excessive usage or security? or rclone or acd_cli? and everyone single 3rd party api is being shut down?

Nope, my Cloud Sync on Synology is still on the 3rd party app page on Amazon Cloud.

I think Amazon might not like the encryption part.. Synology's Cloud Sync doesn't support it.

**Comment by ajkis** [comment]
Who in their right mind would keep data on cloud providers unencrypted.

**Comment by Technifocal** [comment]
@ajkis People storing non-personal/public domain content? I have a decent amount (~500GB, ~25% of what I store on ACD) of unencrypted Youtube videos and the like. The other 1.5TB is encrypted backups, however.

**Comment by notDavid** [comment]
> I put a post in the forum which I'll update as and when I hear more info.

@ncw i think it might be a good idea to create an "ACD status issue" (or forum post) which is locked and only you can post to, so people can subscribe to that and receive notifications/email updates from you, without all the non-info-comments other random users create.

**Comment by SirJMD** [comment]
I just got this from Amazon:

> I understand your concern about using third party apps to access Amazon drive.
> 
> I've checked our Amazon Drive data sources and see that Rclone app access has been revoked and it no longer supports Amazon Drive access. I'm sorry for this disappointment.
> 
> And error with Acd_Cli client access to Cloud Drive is a known issue and our technical team is working on this. We appreciate your patience and understanding as we work to resolve this issue.

**Comment by ajkis** [comment]
It would be funny if rclone is banned and acd_cli gets unbanned

**Comment by ajthemacboy** [comment]
I think I'm moving to G Suite for unlimited Drive storage. My plan is to use a Windows Vultr node to transfer my 420 GBs of files from ACD to GDrive using the Windows clients. I had a friend test the Vultr download speed of a ACD file and he got 80 MB/s. 

I play to use the 100 GB option at $0.06 per hour: https://www.vultr.com/pricing/

That puts my estimate cost at $5.376 + 16 for a Windows license, with an estimated time of 90 hours.

If anyone had a better option than GDrive or Vultr, let me know.

**Comment by gordan-bobic** [comment]
I'm also moving to gdrive.
Multcloud should do what you want for transferring the data. The bandwidth may be slow, though, given that some thousands of other ACD users decided to do the same thing in the past 24 hours.

**Comment by Joshua1337** [comment]
I got myself a new Server from hetzner.de (serverboerse) with 4x4TB did raid0 and put proxmox and a windows 10 vm on it, now im transfering my Data (35TB in total) from ACD to the windows VM and put it then back on gdrive. It will cost me some days of work because it's all encrypted and i have to sync every folder by hand and put it on gdrive but it's the easiest way for me...

**Comment by lloydie2009** [comment]
I had this back from Amazon today... 

Rclone app access has been revoked and it no longer supports Amazon Drive access. I'm sorry for this disappointment.

Acd_Cli client access to Cloud Drive is a known issue and our technical team is working on this. We appreciate your patience and understanding as we work to resolve this issue.

**Comment by Unrepentant-Atheist** [comment]
They banned rclone: https://imgur.com/a/70DbD

**Comment by gordan-bobic** [comment]
Yeah, looks like ACD is gone for good for Linux users. Time to move on.

**Comment by giulianozorzi** [comment]
I've opened a ticket with the customer support and they confirmed the block. they also gave me a 5$ off on next purchase. Maybe if all the users complains they will reconsider the decision or at least release a linux client (...)

**Comment by uncuervo** [comment]
@ncw , could you kindly confirm if there's no option and won't be in the near future?... what do your contacts in amazon say?

Thanks,

**Comment by left1000** [comment]
how much does gsuite unlimited storage cost? googledrive's unlimited tier is 100$ a month which seems totally unreasonable to me.

**Comment by uncuervo** [comment]
@left1000  , Google drive unlimited is only accesible if you have more than 5 users ... if you have less than 5, 1TB limit applies ... :/
then you have the Google storage service, they charge per GB, 20$ month per 1TB ... bloody expensive as well...

**Comment by gordan-bobic** [comment]
Ostensibly, 1TB/user, unlimited with 5+ users.
However - the 1TB limit isn't enforced, I know several people with 2-5TB in their single user gsuite accounts.
Or just band together with another 4 people and get it unlimited even by the terms on the sign-up page.

**Comment by animosity22** [comment]
@uncuervo - you just need 1 person as the 5 user minimum isn't enforced now.

**Comment by uncuervo** [comment]
@Animosity022 , maybe not now..., but who knows what they will do... XD

El 20 may. 2017 1:26, "Animosity022" <notifications@github.com> escribió:

@uncuervo <https://github.com/uncuervo> - you just need 1 person as the 5
user minimum isn't enforced now.

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/ncw/rclone/issues/1417#issuecomment-302832370>, or mute
the thread

**Comment by animosity22** [comment]
@uncuervo - only stating what is accurate for folks at this moment so they can make informed decisions as your statement isn't correct "now" :)

**Comment by uncuervo** [comment]
@Animosity022 , yep, sure, that's fine, but their pricing website says
something different... And I haven't found nothing close to "not min 5
users yet!, you still can be just one user and have unlimited storage,
please come in!" XD  ... Just for the record, where did you get that info
from? :)

El 20 may. 2017 2:26, "Animosity022" <notifications@github.com> escribió:

@uncuervo <https://github.com/uncuervo> - only stating what is accurate for
folks at this moment so they can make informed decisions as your statement
isn't correct "now" :)

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/ncw/rclone/issues/1417#issuecomment-302838044>, or mute
the thread

**Comment by animosity22** [comment]
@uncuervo  - That's what I'm personally using:

http://imgur.com/Sgf7LqR - my GD

http://imgur.com/GTIbUJc - my Billing

**Comment by uncuervo** [comment]
@animosity22 wow! that's great , thanks for the evidence, I hope they don't start to apply their policies whenever is convinient for them (e.g. all the rclone+ACD users going to Gdrive XD ...) thanks a lot for the info :) , I'm working on a WA so far... but if I have no luck now I will consider migrating 100%

**Comment by barnabyc** [comment]
Ok, so if we're all considering transferring our ACD content to Google, what's the best approach to use? What apps actually still work with ACD? Obviously we need something that's automated as many of us have multiple terabytes of data, so pulling it down manually is infeasible. Ideally service-to-service would be ideal. We should create a sticky guide or something for this.

**Comment by thenoahcomputer** [comment]
@barnabyc 
Cloudhq seems to be faster than multcloud, but more expensive.  I'm using both in parallel.  I guess you could maybe find other cloud transfer companies to run as well if you're willing to spend the money.

Further discussion of that probably belongs on the forum or gitter since it's not an rclone issue?

**Comment by poom** [comment]
@animosity22 Which plan do you use it?

I saw G Suite Business  with this remark

> Use Drive to keep all your work files in one secure place with unlimited cloud-based file storage (accounts with fewer than 5 users get 1TB per user). Access your files whenever you need them from your laptop, phone, or tablet.

https://gsuite.google.com/pricing.html?tab_activeEl=tabset-companies

How long you use this plan?

**Comment by ajkis** [comment]
@poom google does not enforce the limit so even with 1 user you will have unlimited space.

**Comment by vampywiz17** [comment]
https://forum.rclone.org/t/rclone-has-been-banned-from-amazon-drive/2314

> Update 2017-05-20
> 
> I have now heard back from Amazon. Rclone has been banned for having the encrypted secrets in the source code.
> 
> I've asked for new credentials so I can build an auth server and get rclone going again - I haven't heard back on that reply yet, but it is plausible we could get rclone running again with Amazon Drive in the not too distant future.
> 
> Thank you all for your patience
> 
> Nick

**Comment by gordan-bobic** [comment]
Any chance of shipping a pre-compiled binary module containing the secrets, and thus avoiding needing an external authentication server?

**Comment by drvdijk** [comment]
@gordan-bobic what would prevent other apps not to use such a binary module?

**Comment by gordan-bobic** [comment]
@drvdijk only the fact that it'd be compiled into the binary and not human readable like it is in the source code.

There are other ACD apps already out there (not naming them for obvious reasons) that haven't been banned that ship with secrets in binaries (they are closed source) and don't use an external authentication server. They may get banned at some point, but having one fewer dependency seems like a good thing right up to the point where it becomes an absolute necessity to introduce it.

**Comment by Technifocal** [comment]
@gordan-bobic what would prevent people from reverse engineering the compiled binary to acquire the secrets? Or simply intercepting the connection (As it's running on my PC, even using HTTPs wouldn't be that hard to read, considering I can read RClone's memory as it decrypts the traffic)? I mean, it's not like it's not done on a daily basis in the video game industry to create external memory modification programs to cheat, or simply bypass and pirate DRM.

I understand Amazon's point of view on this one, they banned ACD_CLI, then people using that software started using another software's authentication credentials to continue using it, while ACD_CLI was not banned for its usage patterns (But instead a concurrency issue that leaked credentials), an app that's banned should be banned, and if the developer wishes to fix the issues and request their credentials back, that's up to the developer.

**Comment by gordan-bobic** [comment]
@Technifocal the fact that very few people know how to do it. Yes, what you describe is indeed entirely viable (not getting into the exact attack vectors, but yes, it does work). But the fact remains there are quite a few other apps still out there that haven't been banned that avoid using authentication servers.

The entire "banning by app" concept is fundamentally flawed, though, because it is entirely ineffective. The sensible thing to do might be to simply alter the rclone back-end and make it fall back on web-scraping - and unless Amazon are going to kill the web interface they might as well give up at that point.

**Comment by JanHellwig** [comment]
> EDIT: Question, what stops someone authenticating with RClone (if they get their credentials back using a server-sided OAUTH system) then using the access token in ~/.rclone.conf to communicate with other software?

I used the token from another software yesterday in order to download my data from ACD with Rclone. Worked perfectly fine except for the fact that refreshing the token didn't work automatically as the other software uses a different format for the expiry date.

**Comment by animosity22** [comment]
@poom - few months now. All the business plans have this and you just have to move a domain over. It took me a few minutes to get it setup.

**Comment by SirJMD** [comment]
Just went to try out Google Drive.. horribly slow :( Getting around 800 kb/s on a 200 MBit/s fiber. What a fail. ACD managed 3-5 MB/s before they shut down access.

**Comment by gordan-bobic** [comment]
Performance varies depending on time of day, where you are and luck. I had ACD struggle to achieve 3Mbit/s at times, and GD is managing to saturate my 200Mbit connection at the moment.

**Comment by SirJMD** [comment]
@gordan-bobic Do you do anything to achieve that? GD is utter garbage for me.

Transferred:   2.274 GBytes (707.319 kBytes/s)
Errors:                 0
Checks:                 0
Transferred:            4
Elapsed time:    56m10.9s

**Comment by calisro** [comment]
@SirJMD that is terrible.  I've always had really good performance from gd.  Must be the end point your connected to.

**Comment by calisro** [comment]
@SirJMD I just maxed my 150 megabit connection with a single transfer to gd. Perhaps there is a way to change your end point.

**Comment by gordan-bobic** [comment]
@SirJMD no special configuration, just straight rclone sync.

**Comment by SirJMD** [comment]
Hmm.. It starts out at 5 MB/s and then just gradually drops down til around 500 kb/s. With ACD I could push about 3 MB/s per thread. Downloading is no problem.. So peering isn't the issue here I guess?

Transferred:   2.274 GBytes (13.883 MBytes/s)
Errors:                 0
Checks:                 0
Transferred:            4
Elapsed time:     2m47.7s

**Comment by mc40638** [comment]
I am been monitoring what has been going on.  As far as the 'cloud' is concerned (I prefer to say google and amazon servers) I, like the rest of you, have gotten some contradictory information.  I am not a power user (have a mere 5 MB/s up from my ISP [Charter, :( ].

I was using ACD to archive files and to store image backups. Since my compressed backups are about 350 GB, a transfer to the cloud (at best) will take 6 days.

I started evaluating GoodSync (GS) (on Windows, though they have a free Linux CLI) and for the most part find the product competent.  The overhead is not insignificant apparently, since when I installed rclone on a VM  and used that to transfer files, I got better throughput. I suspect it is the multiple threads it uses by default for bulk copies/syncs. 

Rclone stopped working (the 429 errors and then authentication errors) which have occupied this thread. At that point GS was still working but about 45% of the transfers failed (error 400 Bad Request). GS support gave me some BS and guaranteed they are using the official API.  Sidebar: GS with GoogleDrive (GD) was working very well, just as it had been with ACD days earlier.

After numerous calls to Amazon Drive support (support is a loose word; they are not technically trained) I spoke to a gentlemen who knew about rclone being banned and about the acd_cli issues. He put me on hold and after about 10 minues came back with some startling news. 

Apparently ACD will stop supporting ANY third party client application.  I asked him to double check. In fact he looked up a closed ticket about the '400' errors in GS and mentioned that the amazon development team reviewed the ticket and said the errors were expected as they are phasing out third part clients. 

He could not tell me if AWS was affected (but it uses the same framework as ACD) .  He speculated that Amazon will eventually roll out a real client  - he had no idea.  I mentioned that AWS users (especially enterprise) might not tolerate this, and he did not have an answer.

I cannot verify any of this, though it is consistent with my observations.

I may exceed the 1TB limit of the GD $10/month offering. I was intrigued by the post of having a single user G Suite account with the observation that Google isnt enforcing the 1TB limit.

My current Google relationship:

I have a FREE google apps account with 10 licenses and 8 users.  I have an upgraded GDRIVE account associated with an account outside of the google apps account domain. (Like G Suite, you must have a domain name).  If I were to convert the goodle apps account to get unlimited storage (8 users x $5/month x 12 months/yr) the annual cost would be $480. WOW! (If they count licenses even more).

I could simply purchase a new domain for about $12 a year, and sign up for G Suite Business (1 user).  Maybe the best way to go. Unfortunately I will have weeks of transferring data from one Google drive to another.  I need to investigate that (GS can be setup to do this, but if buffers the file on my laptop - so I suffer the same horrendous upload speeds).

There seems to be other solutions out there, but this may be the best price point. There is a Canadian outfit SYNC.COM which has attractive pricing for 2TB, however are no clients I have found to access their servers.

Thanks all.

**Comment by fnkr** [comment]
@mc40638 You don't need to purchase an extra domain for that. You can create a new GSuite domain with a subdomain of your existing domain. Also you don't need to download/reupload everything. Just share all the files and folders with the new GSuite account and then transfer ownership of all the files. Unfortunately, you'll have change the ownership of every single file as it won't be sufficient to change the ownership of the shared folder.

**Comment by left1000** [comment]
would it be possible for anyone to donate their official developer api to nick to use to get rclone back open? or is that pointless? because they'll just reban it unless they specifically ask nick to have a new token of his own?

**Comment by niranjanshr13** [comment]
@left1000 try reading this [https://forum.rclone.org/t/guide-rclone-with-acd-using-drivesink-token/2421](drivesink), you can get api to use in amazon drive on rclone; just look at tutorial, you have to get token every 1 hour, because of expiration. 
Side Noe: In token, format with one space (remove all the <enter/return>) in rclone config file.

**Comment by inquam** [comment]
Amazon has indeed not just changed their API requiring changes in rclone but blocked rclone access.
I sent an inquery asking and if this was the case demanded a full refund since my entire use case was mounting with rclone to upload backups taken of my machines to be stored in the cloud.

This was their response:


> Hello,
> 
> I understand your concern about using third party clients or apps to access Amazon drive.
> 
> Rclone app access has been revoked and it no longer supports Amazon Drive access. I'm sorry for this disappointment.
> 
> However, As requested, I've canceled the Unlimited storage subscription and issued full refund in the amount of $59.99.
> 
> Refunds are issued to the payment method used to make the original purchase and usually complete within a few business days. For more information, go to: www.amazon.com/help/aboutrefunds.
> 
> In the meanwhile, try using Amazon Drive Desktop application for Windows / Mac or use Amazon Drive website for other OS devices.
> 
> It is always important for us to hear how customers react to all aspects. Your valuable feedback will help us to improve the selection and service we provide and we're glad you took time to write to us. We do value the feedback of our customers and make every attempt to integrate their comments or suggestions into our stores and procedures.
> 
> We appreciate your patience and understanding as we work to resolve this issue.
> 
> For any further assistance you can reach us 24 hours a day, seven days a week using the following link:
> 
> https://www.amazon.com/gp/help/contact-us/cloud-drive.html
> 
> We look forward to your next visit.
> 
> We'd appreciate your feedback. Please use the buttons below to vote about your experience today.
> 
> Best regards,
> Sadiq
> Amazon.com

**Comment by amaklp** [comment]
@inquam Post it on /r/DataHoarder

**Comment by Fmstrat** [comment]
@ncw Any chance you could lock this issue down for comments from anyone except contributors? It's gotten out of hand from a discussion standpoint (GitHub should be about the fix). That way users can monitor for new comments to get updates (from only you). Thanks for all the hard work!

**Comment by etomm** [comment]
What @niranjanshr13 is really interesting and seems to replace the access token well? Why not following the way of drivesink to solve this case?

**Comment by duke-87** [comment]
I've done the same as @inquam.
They returned my money no questions asked. In less than 25 minutes.

**Comment by newtonseple** [comment]
@ncw Have you considered authenticating through/using go-acd? That project hosts an open auth server at go-acd.appspot.com.

**Comment by ncw** [comment]
I'm going to close this issue now - for the resolution see https://forum.rclone.org/t/rclone-has-been-banned-from-amazon-drive/

**Comment by carlyuanliu** [comment]
It looks like I will not be able to use rclone to upload/download files to/from acd at least for a while.  What a shame.

I have been using rclone to upload large files and folders to acd, as the acd client is not reliable and puts a limit on the file size.  Also, rclone allows me to obtain directory listings for search purposes, as acd does not provide directory listings nor does it do search properly with double byte characters (you would think these functions to be basic for cloud drives).

I thought about ditching acd,  but I can't beat the price.  For example, Google's price for 1TB exceeds acd's Unlimited Storage.  Now that I have accumulated over 2TB on acd, I decide to keep it.  With all its limitations, acd is still good for archiving purposes.

As for my large files (mainly videos), I'll just split them to within limits before uploading.

As to directory listing, I will obtain it before uploading.  I have yet to find a solution to get existing files on acd.  sigh...

**Comment by ajthemacboy** [comment]
This is a development: https://techcrunch.com/2017/06/08/amazon-ends-its-unlimited-cloud-storage-plan/?ncid=rss
